hudi-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadata.java

```diff
@@ -131,6 +132,10 @@ private void initIfNeeded() {
       HoodieFileReader baseFileReader = readers.getKey();
       HoodieMetadataMergedLogRecordReader logRecordScanner = readers.getRight();
 
+      if (baseFileReader == null && logRecordScanner == null) {
+        return Collections.emptyList();
```

Comment:
shouldn't this return Collections.singletonList(Pair.of(key, Option.empty()))

Reply:
I think we should not, the original code returns key based on the metadata data file content, but here we do not have any data file actually.

Reply:
agree. with @danny0405 . that should be for empty partitions say.